src/map/atcommand.c

```diff
@@ -7720,11 +7720,19 @@ ACMD(iteminfo)
 	for (i = 0; i < count; i++) {
 		struct item_data *item_data = item_array[i];
 		if (item_data != NULL) {
+
+			struct item link_item = { 0 };
+			link_item.nameid = item_data->nameid;
+			StringBuf buf;
+			StrBuf->Init(&buf);
+			clif->format_itemlink(&buf, &link_item);
+
 			snprintf(atcmd_output, sizeof(atcmd_output), msg_fd(fd, MSGTBL_ITEMINFO_DETAILS), // Item: '%s'/'%s'[%d] (%d) Type: %s | Extra Effect: %s
-				item_data->name, item_data->jname, item_data->slot, item_data->nameid,
+				item_data->name, StrBuf->Value(&buf), item_data->nameid,
 				itemdb->typename(item_data->type),
 				(item_data->script == NULL) ? msg_fd(fd, MSGTBL_ITEMINFO_NONE) : msg_fd(fd, MSGTBL_ITEMINFO_WITH_SCRIPT) // None / With script
 			);
+			StrBuf->Destroy(&buf);
```

Comment:
The StringBuf could be `Init()`ialized and `Destroy()`ed outside the for loop, and only `Clear()`ed here, in order to avoid unnecessary memory reallocations

Reply:
Thanks, 
Performed those changes and tested it.